src/test/java/org/opensearch/security/filter/SecurityRestFilterTests.java

```diff
@@ -300,4 +303,79 @@ public void testAllowlistedApiWithoutExtraSlash() throws Exception {
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testWithoutHasPermissionCheckParam() throws Exception {
+        setup();
+
+        // ALLOWLIST GET /_cluster/health
+        rh.keystore = "restapi/kirk-keystore.jks";
+        rh.sendAdminCertificate = true;
+        response = rh.executePutRequest(
+            "_plugins/_security/api/allowlist",
+            "{\"enabled\": true, \"requests\": {\"/_cluster/health\": [\"GET\"]}}",
+            nonAdminCredsHeader
+        );
+
+        rh.sendAdminCertificate = false;
+
+        response = rh.executeGetRequest("_cluster/health", nonAdminCredsHeader);
+        assertThat(response.getStatusCode(), equalTo(HttpStatus.SC_OK));
+        assertFalse(response.getBody().contains("\"accessAllowed\""));
+        assertFalse(response.getBody().contains("\"missingPrivileges\""));
+
+    }
+
+    /**
+     * Tests that the has_permission_check param works correctly.
+     * When has_permission_check=true is added to a request, returns
+     * whether the request would be allowed or Denied, without actually executing the request.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testHasPermissionCheckParam() throws Exception {
+        setup();
+
+        // ALLOWLIST GET /_cluster/health
+        rh.keystore = "restapi/kirk-keystore.jks";
+        rh.sendAdminCertificate = true;
+        response = rh.executePutRequest(
+            "_plugins/_security/api/allowlist",
+            "{\"enabled\": true, \"requests\": {\"/_cluster/health\": [\"GET\"]}}",
+            nonAdminCredsHeader
+        );
+
+        rh.sendAdminCertificate = false;
+        response = rh.executeGetRequest(
+            "_cluster/health?" + ConfigConstants.SECURITY_HAS_PERMISSION_CHECK_PARAM + "=true",
+            nonAdminCredsHeader
+        );
+        assertThat(response.getBody(), response.getStatusCode(), equalTo(HttpStatus.SC_OK));
+        assertTrue(response.getBody().contains("\"accessAllowed\":"));
+        assertTrue(response.getBody().contains("\"missingPrivileges\":"));
+
+        rh.sendAdminCertificate = true;
+        response = rh.executePutRequest(
+            "_plugins/_security/api/allowlist",
+            "{\"enabled\": true, \"requests\": {\"/_search\": [\"GET\"]}}",
+            adminCredsHeader
+        );
+
+        rh.sendAdminCertificate = false;
+        response = rh.executeGetRequest("_search?" + ConfigConstants.SECURITY_HAS_PERMISSION_CHECK_PARAM + "=false", nonAdminCredsHeader);
+        assertThat(response.getBody(), response.getStatusCode(), equalTo(HttpStatus.SC_OK));
+        assertFalse(response.getBody().contains("\"accessAllowed\":"));
+        assertFalse(response.getBody().contains("\"missingPrivileges\":"));
+
+        response = rh.executeGetRequest("_cat/nodes?" + ConfigConstants.SECURITY_HAS_PERMISSION_CHECK_PARAM + "=true", nonAdminCredsHeader);
+        assertThat(response.getBody(), response.getStatusCode(), equalTo(HttpStatus.SC_FORBIDDEN));
```

Comment:
Why is Forbidden expected here instead of the response saying allowed is false with the list of missing permissions?